datastore/datastore.go

```diff
@@ -24,10 +24,12 @@ import (
 const (
 	errMsgReadingFromStoreF = "Error reading %q from datastore."
 
-	// This number should be kept quite a lot higher than the number of
+	// This number 30 should be kept quite a lot higher than the number of
 	// expected state storage operations, which is usually roughly
-	// equivalent to the number of state transitions.
-	MaximumStateDataStoreCount = 30
+	// equivalent to the number of state transitions. 40 is added as an
```

Comment:
10 is added as an extra buffer?

Reply:
But we are adding 40..? Doesn't the comment explain why we add 40?

Reply:
It just sounded a bit off, with the `this number 30` in the previous sentence, and then 40 is added. But no biggie.

Reply:
I suppose I could have summed them and added that, but since the numbers 30 and 40 come from different sources, I opted for keeping them separate for clarity. At some point in the future, maybe we can get rid of the 40 if we fix the underlying problem with the retry-state-transitions.

Reply:
:+1: